### ISSUE TITLE
Generate additional k8s/openshift resource: configmap, secret...

### ISSUE THREAD

**Issue by aureamunoz** [dekorateio/dekorate#335]
Review/Enhance the logic of dekorate to offer the possibility to copy additional yml resources files such as ConfigMap, Secret, ... to the target dekorate folder

**Comment by Sgitario** [comment]
@aureamunoz I think this is already supported (I've seen a session file reader), but I can't find how to use it in the documentation. Can you confirm what is needed here? If some implementation is needed or just document how to use it.

**Comment by aureamunoz** [comment]
I'm not sure about what was wanted with this ticket, it's very old....maybe @iocanel or @cmoulliard can say?

**Comment by cmoulliard** [comment]
The idea here was to generate a ConfigMap or a Secret part of the process to generate k8s manifest using by example as convention using the application properties

```
dekorate.kubernetes.configmap.name=my_configmap
dekorate.kubernetes.configmap.name.foo=bar
dekorate.kubernetes.configmap.name.another_foo=another_bar
```
**Remark**: If the object is a secret, then the data will be base64 encoded

**Comment by Sgitario** [comment]
The `SimpleFileReader` was done for a similar purpose (added in https://github.com/dekorateio/dekorate/commit/fd57866a411da02deb1d6292c186e79504888970). 
The idea of the `SimpleFileReader` was to read an input path and incorporate the found resources into the target folder along to the other auto-generated files.
At the moment, this feature is not used in Dekorate, but it is [in Quarkus](https://github.com/quarkusio/quarkus/blob/77a14d4cd7ac4962bef79929eeaf064371d01e25/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java#L130). 
From my point of view and to avoid to over-complicate the generation logic within Dekorate, I would prefer to use this `SimpleFileReader` approach also in Dekorate instead of generating resources from application properties (which is much more complex and might conflict with other existing properties).
Wdyt? @iocanel @aureamunoz @cmoulliard

**Comment by aureamunoz** [comment]
>  From my point of view and to avoid to over-complicate the generation logic within Dekorate, I would prefer to use this `SimpleFileReader` approach also in Dekorate instead of generating resources from application properties (which is much more complex and might conflict with other existing properties). Wdyt? @iocanel @aureamunoz @cmoulliard

Yes, I agree with this. Besides, currently we are generating configmap and secrets part of the process, there is an [example](https://github.com/dekorateio/dekorate/blob/1dfaea7d36b543a0e610090a32abb60e8eae4d9d/examples/kubernetes-with-configmap-volume-example/src/main/java/io/dekorate/example/Main.java#L23) that showcases it. The idea here is to add eventual yaml resources, no?
